Wincenttjoi/CS3223-DBMS-Project
# Lab2 index dev

Note:
- `IndexRetrievalTest.java` and `indexUpdateTest.java` only works in network db (make sure run server)
- `SimpleIJ.java` uses hardcoded select, just need to change the query in this file (line 34) for future testing purposes
- Download and check the lab2 report in root folder

Steps to test:
1. Run server network
2. Delete network db if any, then rerun `CreateStudentDB.java`
3. You may put a System.out.println in `open:IndexInfo` each for case hash and case btree, see below
![Screenshot 2022-02-03 at 4 56 51 PM](https://user-images.githubusercontent.com/46394439/152311082-6bba2ca8-52dc-444a-aaf3-1208168fe2dc.png)
4. Run `IndexRetrievalTest` and `IndexUpdateTest`, observe output -> check if your System.out.println is on console
5. Go to `CreateStudentDB` and change using to the other index type btree/hash, redo step 2-4 and check the console should print the other case  ![Screenshot 2022-02-03 at 4 59 45 PM](https://user-images.githubusercontent.com/46394439/152311540-8357832e-4d7f-4588-9d41-ab840ba05f16.png)
